blog/_posts/2025-04-21-vm-generations-and-aks.md

```diff
@@ -1 +1,106 @@
+| Latest v6 VM Support | v6 VMs support Generation 2 VMs | v6 VMs do **NOT** support Generation 1 VMs |
+| [Trusted Launch](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch) | Can enable Trusted Launch, which includes protections like virtual Trusted Platform Module (vTPM) | Can **NOT** enable Trusted Launch |
+| [NVMe Interface Support](https://learn.microsoft.com/en-us/azure/virtual-machines/nvme-overview) | Supports NVMe disks, which requires NVM-enabled Generation 2 image | Does **NOT** support NVMe disks |
+
+### Implications for your Virtual Machines
+If you are already running on Generation 2 VMs, you are all set to deploy on the majority of Azure VMs, including the latest v6 VMs. You can also enable Trusted Launch and the NVMe Interface. 
+
+If you are running on a Generation 1 VMs, you can continue running on most v5 and older Azure VMs. Migration to Generation 2 VMs is recommended, especially if any of the following requirements apply: 
+- You require Trusted Launch for your workloads 
+- You require NVMe interface for your workloads
+- You want/need to migrate to the latest v6 VMs 
+
+## VM Generation Support on AKS
+
+### Generation 2 Default
+
+AKS supports both Generation 1 and 2 VMs with all operating systems on AKS. Generation 2 VMs support additional features Generation 1 VMs do not, such as increased memory, virtualized persistent memory, or a UEFI-based boot architecture. The VM size and operating system that you select when creating an AKS node pool determines which VM Generation you will use. 
+- When creating Linux node pools on AKS, the default will be a Generation 2 VM unless the VM size does not support it. 
+- When creating Windows Server 2025 node pools on AKS, the default will be a Generation 2 VM unless the VM size does not support it.  
+- When creating Windows Server 2019 and Windows Server 2022 node pools on AKS, the default will be Generation 1 VM unless the VM size does not support it. To use a Generation 2 VM, you must add `--aks-custom-headers UseWindowsGen2VM=true` during node pool creation. 
+
+For more information on Generation 2 default behavior on AKS, see [AKS documentation](https://learn.microsoft.com/en-us/azure/aks/generation-2-vm?tabs=linux-node-pool).
+
+For a list of supported VM sizes for Generation 1 and Generation 2, please refer to the table on [this page](https://learn.microsoft.com/en-us/azure/virtual-machines/generation-2#generation-2-vm-sizes).
+
+### Generation 1 VM Retirements
+When a VM size or series reaches its retirement date, the VM will be deallocated. VM deallocation means that your AKS node pool may experience breakage.
+
+If you would like to confirm whether your Gen 1 VM sizes are retired or are being retired, please search in the [Azure Updates](https://azure.microsoft.com/en-us/updates) 
```

Comment:
```suggestion
If you would like to confirm whether your Gen 1 VM sizes are retired or are being retired, please search in the [Azure Updates](https://azure.microsoft.com/updates) 
```

Reply:
I think we can add more here:
- Azure Retirement workbook (Learn more about service retirements that may impact your resources in the [Azure Retirement Workbook](https://aka.ms/ServicesRetirementWorkbook). Please note that retirements may not be visible in the workbook for up to two weeks after being announced.)
- List of retired VM sizes: https://learn.microsoft.com/azure/virtual-machines/sizes/retirement/retired-sizes-list